File: leetcode/Python/search-in-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def searchBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        sol, cur = None, root
        while cur:
            if cur.val>val:
                cur = cur.left
            elif cur.val<val:
                cur = cur.right
            else:
                sol = cur
                break
        return sol

File: leetcode/C++/search-in-a-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* searchBST(TreeNode* root, int val) {
        TreeNode *sol = nullptr;
        for(TreeNode* cur=root; cur; ){
            if(cur->val>val) cur=cur->left;
            else if(cur->val<val) cur=cur->right;
            else {
                sol=cur;
                break;
            }
        }
        return sol;
    }
};